# Author: Caspar van Leeuwen, SURF
# If the CPU was idle (i.e. cpuPercent=0), the display showing CPU usage wouldn't update,
# because of a faulty logic. This patch fixes that.
# See https://github.com/jupyter-server/jupyter-resource-usage/pull/147
diff -Nru jupyter-resource-usage-0.6.2.orig/jupyter_resource_usage/static/main.js jupyter-resource-usage-0.6.2/jupyter_resource_usage/static/main.js
--- jupyter-resource-usage-0.6.2.orig/jupyter_resource_usage/static/main.js	2022-09-30 19:02:36.892711996 +0200
+++ jupyter-resource-usage-0.6.2/jupyter_resource_usage/static/main.js	2022-09-30 19:02:59.994787990 +0200
@@ -74,7 +74,7 @@
 
                 // Handle CPU display
                 var cpuPercent = data['cpu_percent'];
-                if (cpuPercent) {
+                if (cpuPercent !== undefined) {
                     // Remove hide CSS class if the metrics API gives us a CPU percent to display
                     $('#jupyter-resource-usage-display-cpu').removeClass('jupyter-resource-usage-hide');
                     var maxCpu = data['cpu_count'];
